posthog/api/test/test_insight.py

```diff
@@ -152,6 +153,23 @@ def test_basic_results(self):
                 ],
             )
 
+        @snapshot_postgres_queries
+        def test_insights_does_not_nplus1(self):
```

Comment:
I don't quite see how this test applies to this issue. `self.assertNumQueries(6)` passes both with `prefetch_related` and `select_related`, while the difference between the snapshots of these is just these joins – but this does not seem to test that with `prefetch_related` the related rows are cache.

Reply:
The number of queries is the same, but the queries aren't. `select_related` tries to also do a left-join for the two tables, resulting in a very wide queryset. Not sure why it does an extra query on top of that though.

If you have many-to-many relations then trying to fetch things in one query is often worse than N queries - e.g. in this case if the same user creates all insights we still fetch the persons data LIMIT times in the result.

Reply:
Still something wrong with the test... moment!

Reply:
And resolved! Hopefully this makes more sense now.